p2p/dandelion-tower/src/router.rs

```diff
@@ -165,27 +171,41 @@ where
             match ready!(self
                 .outbound_peer_discover
                 .as_mut()
-                .poll_discover(cx)
-                .map_err(DandelionRouterError::OutboundPeerDiscoverError))
+                .try_poll_next(cx)
+                .map_err(DandelionRouterError::OutboundPeerStreamError))
             .ok_or(DandelionRouterError::OutboundPeerDiscoverExited)??
             {
-                Change::Insert(key, svc) => {
+                OutboundPeer::Peer(key, svc) => {
                     self.stem_peers.insert(key, svc);
                 }
-                Change::Remove(key) => {
-                    self.stem_peers.remove(&key);
+                OutboundPeer::Exhausted => {
+                    tracing::warn!("Failed to retrieve enough outbound peers for optimal dandelion++, privacy may be degraded.");
+                    return Poll::Ready(Ok(()));
                 }
```

Comment:
This seems... not okay? What does `monerod` here?

Reply:
monerod does the same thing, there isn't really much more we can do here, if we have no outbound peers it is better for transactions to get to the network with less privacy than to be dropped completely.

Even if you fluff immediately you still have some sort of "herd immunity", it shouldn't be easy to tell this was the case.